src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.documentElement.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  // eslint-disable-next-line prefer-promise-reject-errors
+  window.setTimeout(() => reject('First promise was rejected'), 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  document.documentElement.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.documentElement.addEventListener('contextmenu', () => {
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  document.documentElement.addEventListener('click', () => {
+    leftClick = true;
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
+    }
+  });
+
+  document.documentElement.addEventListener('contextmenu', () => {
+    rightClick = true;
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
```

Comment:
The third promise resolves when both 'click' and 'contextmenu' events have occurred. However, it will resolve twice if both events occur, due to the logic in both event listeners. Ensure this is the desired behavior.